server/hooks.go

```diff
@@ -19,15 +18,12 @@ func (p *Plugin) UserHasJoinedTeam(c *plugin.Context, teamMember *model.TeamMemb
 	}
 
 	for _, message := range p.getWelcomeMessages() {
-		var teamNamesArr = strings.Split(message.TeamName, ",")
-		for _, name := range teamNamesArr {
-			tn := strings.TrimSpace(name)
-			if tn == data.Team.Name {
-				if data.User.IsGuest() && !message.IncludeGuests {
-					continue
-				}
-				go p.processWelcomeMessage(*data, *message)
-			}
+		if data.User.IsGuest() && !message.IncludeGuests {
+			continue
+		}
+
+		if message.TeamName == data.Team.Name {
+			go p.processWelcomeMessage(*data, *message)
```

Comment:
This could be made a bit easier to read

```suggestion
		teamNameMatches := message.TeamName == data.Team.Name
		restrictGuest := data.User.IsGuest() && !message.IncludeGuests
		if teamNameMatches && !restrictGuest {
			go p.processWelcomeMessage(*data, *message)
```

or

```suggestion
		if data.User.IsGuest() && !message.IncludeGuests {
			continue
		}

		if message.TeamName != data.Team.Name {
			continue
		}

		go p.processWelcomeMessage(*data, *message)
```

@trilopin What do you think?